File: ideas/procpp/mixin_handler/tests/ws.php
<?php
require(__DIR__ . "/../build/php/ReactHandler.php");
class PhpCallback extends WebSocketCallback {
    function run($m, $ws)
    {
        if ($m->opcode == WebSocket::OPCODE_TEXT_FRAME) $ws->send($m->client, $m->payload);
    }
}

$callback = new PhpCallback();
$callback->thisown = 0;

$s=new WebSocketServer();
$s->on('message', $callback);
$s->listen(8082);
$s->event_loop();

File: ideas/procpp/mixin_handler/tests/ws.py
import sys
sys.path.append('build/python/')

import ReactHandler

class PythonCallback(ReactHandler.WebSocketCallback):
    def run(self, m, ws):
            if m.opcode == ReactHandler.WebSocket.OPCODE_TEXT_FRAME:
                ws.send(m.client, m.payload)

callback = PythonCallback().__disown__()

s = ReactHandler.WebSocketServer()
s.on('message', callback)
s.listen(8082)
s.event_loop()

